easybuild/tools/job/gc3pie.py

```diff
@@ -0,0 +1,195 @@
+##
+"""Interface for submitting jobs via GC3Pie"""
+
+
+import os
+import time
+
+try:
+    import gc3libs
+    from gc3libs import Application, Run, create_engine
+    from gc3libs.core import Engine
+    from gc3libs.quantity import hours as hr
+    from gc3libs.workflow import DependentTaskCollection
+    HAVE_GC3PIE = True
+except ImportError:
+    HAVE_GC3PIE = False
+
+from easybuild.tools.build_log import print_msg
+from easybuild.tools.job import JobServer
+
+from vsc.utils import fancylogger
+
+# inject EasyBuild logger into GC3Pie
+gc3libs.log = fancylogger.getLogger('gc3pie', fname=False)
+# make handling of log.error compatible with stdlib logging
+gc3libs.log.raiseError = False
+
+
+# eb --job --job-backend=GC3Pie
+class GC3Pie(JobServer):
+    """
+    Use the GC3Pie__ framework to submit and monitor compilation jobs.
+
+    In contrast with acessing an external service, GC3Pie implements
+    its own workflow manager, which means ``eb --job
+    --job-backend=GC3Pie`` will keep running until all jobs have
+    terminated.
+
+    .. __: http://gc3pie.googlecode.com/
+    """
+
+    USABLE = HAVE_GC3PIE
+
+    def begin(self):
+        """
+        Start a bulk job submission.
+
+        Removes any reference to previously-submitted jobs.
+        """
+        self._jobs = DependentTaskCollection()
+
+    def make_job(self, script, name, env_vars=None, hours=None, cores=None):
+        """
+        Create and return a job object with the given parameters.
+
+        First argument `server` is an instance of the corresponding
+        `JobServer` class, i.e., a `GC3Pie`:class: instance in this case.
+
+        Second argument `script` is the content of the job script
+        itself, i.e., the sequence of shell commands that will be
+        executed.
+
+        Third argument `name` sets the job human-readable name.
+
+        Fourth (optional) argument `env_vars` is a dictionary with
+        key-value pairs of environment variables that should be passed
+        on to the job.
+
+        Fifth and sith (optional) arguments `hours` and `cores` should be
+        integer values:
+        * hours must be in the range 1 .. MAX_WALLTIME;
+        * cores depends on which cluster the job is being run.
+        """
+        extra_args = {
+            'jobname': name, # job name in GC3Pie
+            'name':    name, # same in EasyBuild
+        }
+        if env_vars:
+            extra_args['environment'] = env_vars
+        if hours:
+            extra_args['requested_walltime'] = hours*hr
+        if cores:
+            extra_args['requested_cores'] = cores
+        return Application(
+            # arguments
+            ['/bin/sh', '-c', script],
+            # no need to stage files in or out
+            inputs=[],
+            outputs=[],
+            # where should the output (STDOUT/STDERR) files be downloaded to?
+            output_dir=('%s/easybuild-jobs/%s' % (os.getcwd(), name)),
```

Comment:
this location should be configurable. `getcwd` can be a default, but it's not to be trusted.
also, use 
```python
os.path.join(os.getcwd(), ''easybuild-jobs', name)
```